main.tf

```diff
@@ -113,7 +117,7 @@ resource "aws_instance" "default" {
 }
 
 resource "aws_eip" "default" {
-  count    = "${var.associate_public_ip_address && var.instance_enabled ? 1 : 0}"
+  count    = "${var.associate_public_ip_address && local.instance_count ? 1 : 0}"
```

Comment:
`var.associate_public_ip_address && local.instance_count` means `bool && int`.
Should be replaced with corresponding `locals` too.

Reply:
We shouldn't mix `bool` and `int` into a single expression. Use `bool` instead of `int`.

Reply:
Nevermind, [Booleans](https://www.terraform.io/docs/configuration/variables.html#booleans) says, that `int` `1` will be interpolated as `true`.